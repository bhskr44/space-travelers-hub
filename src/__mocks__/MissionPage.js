import React from 'react';
import MissionList from '../components/mission/MissionList';

const MissionPage = ({ missionsObj }) => {
  return (
    <div>
      <MissionList missions={missionsObj} />
    </div>
  );
};

export default MissionPage;
