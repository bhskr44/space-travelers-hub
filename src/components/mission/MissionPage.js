import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import MissionList from './MissionList';
import { getMissions } from '../../redux/api';

const MissionPage = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions.missions);

  useEffect(() => {
    if (missions.length === 0) {
      dispatch(getMissions());
    }
  }, [dispatch, missions.length]);

  return (
    <div>
      <MissionList missions={missions} />
    </div>
  );
};

export default MissionPage;
