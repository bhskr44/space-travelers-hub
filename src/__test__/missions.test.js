import { render, screen, fireEvent } from '@testing-library/react';
import * as reactRedux from 'react-redux';
import MissionPage from '../__mocks__/MissionPage';

jest.mock('react-redux');

const mockDispatch = jest.fn();
const mockUseDispatch = jest.spyOn(reactRedux, 'useDispatch');


beforeAll(() => {
  reactRedux.useDispatch = jest.fn().mockImplementation(() => mockDispatch);
});

describe('Test for Missions Page', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render the Missions Page', async () => {
    const missionsObj = [
      {
        mission_name: 'Thaicom',
        mission_id: '9D1B7E0',
        manufacturers: ['Orbital ATK'],
        payload_ids: ['Thaicom 6', 'Thaicom 8'],
        wikipedia: 'https://en.wikipedia.org/wiki/Thaicom',
        website: 'http://www.thaicom.net/en/satellites/overview',
        twitter: 'https://twitter.com/thaicomplc',
        description: `Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology`,
      },
    ];

    const tree = render(<MissionPage missionsObj={missionsObj} />);
    expect(tree).toMatchSnapshot();
  });

 
});
