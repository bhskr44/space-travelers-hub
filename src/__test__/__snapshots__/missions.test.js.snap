// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Missions Page should render the Missions Page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="rocket-list"
        >
          <table
            class="mission-table"
          >
            <thead>
              <tr>
                <th
                  class="td-name"
                >
                  <h3>
                    Mission
                  </h3>
                </th>
                <th
                  class="td-description"
                >
                  <h3>
                    Description
                  </h3>
                </th>
                <th
                  class="td-status"
                >
                  <h3>
                    Status
                  </h3>
                </th>
                <th
                  class="td-button"
                >
                  <h3>
                     
                  </h3>
                </th>
              </tr>
            </thead>
            <tr>
              <td>
                <h3>
                  Thaicom
                </h3>
              </td>
              <td
                class="table-description"
              >
                Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology
              </td>
              <td>
                <button
                  class="not-member"
                  type="button"
                >
                  NOT A MEMBER
                </button>
              </td>
              <td>
                <button
                  class="join-mission"
                  type="button"
                >
                  Join Mission
                </button>
              </td>
            </tr>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="rocket-list"
      >
        <table
          class="mission-table"
        >
          <thead>
            <tr>
              <th
                class="td-name"
              >
                <h3>
                  Mission
                </h3>
              </th>
              <th
                class="td-description"
              >
                <h3>
                  Description
                </h3>
              </th>
              <th
                class="td-status"
              >
                <h3>
                  Status
                </h3>
              </th>
              <th
                class="td-button"
              >
                <h3>
                   
                </h3>
              </th>
            </tr>
          </thead>
          <tr>
            <td>
              <h3>
                Thaicom
              </h3>
            </td>
            <td
              class="table-description"
            >
              Thaicom is the name of a series of communications satellites operated from Thailand, and also the name of Thaicom Public Company Limited, which is the company that owns and operates the Thaicom satellite fleet and other telecommunication businesses in Thailand and throughout the Asia-Pacific region. The satellite projects were named Thaicom by the King of Thailand, His Majesty the King Bhumibol Adulyadej, as a symbol of the linkage between Thailand and modern communications technology
            </td>
            <td>
              <button
                class="not-member"
                type="button"
              >
                NOT A MEMBER
              </button>
            </td>
            <td>
              <button
                class="join-mission"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
